pkg/core/blockchain.go

```diff
@@ -556,6 +556,15 @@ func (bc *Blockchain) storeBlock(block *block.Block) error {
 		return err
 	}
 
+	if block.Index > 0 {
+		systemInterop := bc.newInteropContext(trigger.System, cache, block, nil)
+		v := SpawnVM(systemInterop)
+		v.LoadScriptWithFlags(bc.contracts.GetPersistScript(), smartcontract.AllowModifyStates|smartcontract.AllowCall)
+		if err := v.Run(); err != nil {
+			bc.log.Warn("can't persist native contracts", zap.Error(err))
```

Comment:
I think we should fail persisting the block then. And there also should be a wrapper cached DAO for these cases. In general, it's not that different from transaction execution handling.

Reply:
Fixed.

Reply:
What do you mean by `wrapper cached DAO`? We use already created cache and `interop.NewContext` takes care about wrapping.

Reply:
OK.

Reply:
But I don't see it being persisted though.

Reply:
Oh, yeah, fixed.